Add content about LSP8 TokenId Format + Metadata differences (#766)
lukso-network/docs: docs/standards/tokens/LSP8-Identifiable-Digital-Asset.md

This standard was based on **[ERC721](https://eips.ethereum.org/EIPS/eip-20)** and **[ERC1155](https://eips.ethereum.org/EIPS/eip-777)** with additional features mentioned below:

### Format of TokenIds

The current NFT standards such as **[ERC721](https://eips.ethereum.org/EIPS/eip-721)** and **[ERC1155](https://eips.ethereum.org/EIPS/eip-1155)** **lack asset representation** as they define the tokenIds **as Numbers** `(uint256)`. Each token from the NFT collection will be defined and queried based on this tokenId, which is normally incremental.

![ERC721 TokenIds Representation](/img/standards/lsp8/erc721-tokenIds.jpeg)

> **[LSP8-IdentifiableDigitalAsset](#)** define the tokenIds as `bytes32`. The choice of format for the **`bytes32 tokenId`** allows a wide variety of applications including numbers, string names, contract addresses, unique bytes identifiers (e.g: serial numbers) or hashed values.

Reviewer: ```suggestion
**[LSP8-IdentifiableDigitalAsset](#)** defines the `tokenId` element as `bytes32` to allow a wide variety of data, including numbers, string names, smart contract addresses, byte identifiers for serial numbers, or even hashed values.
```

I suggest not using double-styling.

> Finally a LSP8 Collection contract can also contains NFTs of different formats, illustrated under the **Mixed Formats tab** below. For instance all the NFTs by default are of format `number`, but certain NFTs in the collection could be more complex and represented as their own smart contract. In this example use case, the tokenId format will be **`Mixed` with default as `address`** (= `102`).

Reviewer: ```suggestion
Finally, a LSP8 Collection contract can also contain NFTs of different formats, illustrated under the **Mixed Formats tab** below. For instance, all the NFTs by default are of the `number` format, but certain NFTs in the collection could be more complex and represented as their own smart contract. In this example use case, the tokenId format will be **`Mixed` with default as `address`** (Value `102`).
```

### `force` mint and transfer

:::success Recommendation

> It is advised to set the `force` boolean to `false` by default when transferring or minting tokens to prevent sending them to the wrong address. For instance, if the wrong address was pasted by mistake by the user in the input field of a dApp.

Reviewer: ```suggestion
When transferring or minting tokens, setting the `force` boolean to `false` is advised to limit token transactions to [ERC725 Accounts](..//universal-profile/lsp0-erc725account.md). This will ensure tokens won't be lost when sent to the wrong EOA or contract address.
```

Author: Addressed but wrote an alternative sentence. I prefer phrasing it the other way around, as the word "limit" looses the emphasis on the purpose of the `force` parameter.

Implementing the **[LSP1-UniversalReceiver](../generic-standards/lsp1-universal-receiver.md)** standard will give a sign that the contract knows how to handle the tokens received.

## LSP8 Collection vs TokenId Metadata

> The LSP8 standard offers ways to set metadata for the whole LSP8 collection as well as metadata per specific token Id. This is done using the same `LSP4Metadata` data key, but different functions.

Reviewer: ```suggestion
The LSP8 standard offers ways to set metadata for the whole LSP8 collection or a specific token ID. This is done using the same `LSP4Metadata` data key but with different functions described below:
```

> The generic metadata and information related to the whole LSP8 collection can be updated using [`setData(bytes32,bytes)`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdata) or [`setDataBatch(bytes[],bytes[])`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdatabatch), passing the `bytes32` data key of [`LSP4Metadata`](./LSP4-Digital-Asset-Metadata.md#lsp4metadata) as the first argument.
> 
> Alternatively, the generic information and metadata specific to a `tokenId` can be set using the [`setDataForTokenId`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdatafortokenid) or [`setDataBatchForTokenIds(bytes32[],bytes32[],bytes[])`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdatabatchfortokenids) functions.
> 
> :::

Reviewer: ```suggestion
- The generic metadata and information related to the whole LSP8 collection can be updated using [`setData(bytes32,bytes)`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdata) or [`setDataBatch(bytes[],bytes[])`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdatabatch), passing the `bytes32` data key of [`LSP4Metadata`](./LSP4-Digital-Asset-Metadata.md#lsp4metadata) as the first argument.

- Alternatively, the generic information and metadata specific to a `tokenId` can be set using the [`setDataForTokenId`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdatafortokenid) or [`setDataBatchForTokenIds(bytes32[],bytes32[],bytes[])`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdatabatchfortokenids) functions.

```

I would suggest using bullet points. I also was a `:::` left over
